Activation params

arminsabouri opened the pull request.

arminsabouri committed:
feature(edh): version 1 of edh with voting bits

arminsabouri committed:
docs(abci): comment about why version is in the middle of the ndd datastructure

arminsabouri committed:
feature(ndd): add version 1 with voting bits

arminsabouri committed:
WIP - define activation activation params

arminsabouri committed:
test(contracts): fix broken event emit check

arminsabouri committed:
docs(contracts): add README.md

arminsabouri committed:
chore(net): remove unused `NetworkFrostEvent` struct

arminsabouri committed:
fix(net): ensure we have a connection with a peer when receving messages

arminsabouri committed:
chore(authority): missing doc tests for test_utils

arminsabouri committed:
chore(net): remove unused `PeerConfirmed` bool

arminsabouri committed:
docs(btc_server): update README

SatwikPrabhu committed:
Chore: Appropriate error for botanix consensus pkg

Pass error directly instead of string

Fix crate check eror

arminsabouri committed:
fix(net): frost identifiers for signing payload should be calculated on session established

arminsabouri committed:
chore(net): remove unused utxo message and response type

arminsabouri committed:
chore(poa): remove unused execute fetched block

arminsabouri committed:
chore(evm): clean up block validation error variants

arminsabouri committed:
refactor(poa): return new err when we fail to fetch block producer address

arminsabouri committed:
refactor(poa): remove extra params to complete header

scottmillner committed:
bug-fix(test-suite): don't use features flag when spawning btc-servers, only use for poa nodes

arminsabouri committed:
test: update rpc error string in assert

scottmillner committed:
bug-fix(test-suite): compile conflicting_input feature in integration test flow

arminsabouri committed:
fix(btc_server): do not create dust change output

arminsabouri committed:
fix(btc_server): fix finding pegout by using pegout spk

using amount wont work b/c we take the fee out of the pegout amount

arminsabouri committed:
chore: update frost lib commit to c018ae388c7a10fb9534391b8d09cb07230b78c4

arminsabouri committed:
fix(primitives): use workspace frost taproot crate

arminsabouri committed:
refactor: consolidate all taproot spk generation to btc_wallet crate

arminsabouri committed:
ci: remove validate pr name job

arminsabouri committed:
fix(poa): remove amount check when checking for pegouts in psbt

scottmillner committed:
ci(contracts): run workflow on pr

scottmillner commented on crates/consensus/authority/src/comet_bft/non_deterministic_data.rs:
```diff
@@ -20,9 +20,11 @@ pub(crate) enum NonDeterministicDataDeserializeError {
 pub(crate) struct NonDeterministicData {
-    pub(crate) version: u16,
     pub(crate) bitcoin_block_hash: bitcoin::hash_types::BlockHash,
     pub(crate) aggregated_public_key: secp256k1::PublicKey,
```
nice comment! remember when we hit this way back

arminsabouri committed:
chore(header_ext): remove redundant todos

arminsabouri committed:
chore(import): remove unneeded todo comments

arminsabouri committed:
chore: clean up poa entry

arminsabouri committed:
chore(btc_wallet): remove unneeded todo comment

arminsabouri committed:
chore(abci): remove unsued rpc mod

arminsabouri committed:
chore(consensus): use botanix testnet chain id instead of hardcoded value

arminsabouri committed:
chore: update frost lib commit to b4c5ab017bb0d182f3c56fd80b95d6375a4bfeea

scottmillner committed:
test(btc-server): refactor validate_psbt_by_output() to reflect latest changes

arminsabouri committed:
test: update `BITCOIND_URL` to use 127.0.0.1 ip

arminsabouri committed:
ci: disable slither and e-burger reports

rwlockbg committed:
feat(repo): Added none-blocking commit message sending

arminsabouri committed:
refactor: log bip34 missing error

arminsabouri committed:
fix(btc_server): break out getting height into a seperate step

arminsabouri committed:
chore(primitives): fix misc. warnings in botanix utils

arminsabouri committed:
test: abstract out min coinbase confs to a pub const

arminsabouri committed:
test: re-enable the 80-20 split check

scottmillner committed:
fix(abci): handle errors when configuring payload builder

arminsabouri committed:
chore(btc_server): remove unused fee conversion const

scottmillner committed:
feature(abci): refactor test_commit()

arminsabouri committed:
refactor(btc_server): create sub modules for server components

This work creates sub modules for:
* Coordinator
* Signer
* Database
* Pegout Scheduler
* Utils

The idea here is to leave signing and FROST coordinating as stateless
functions while the grpc is the main binary here that uses these
primitives to build the GRPC application

test - int tests

SatwikPrabhu committed:
test: finish test coverage for botanix peg contract

SatwikPrabhu committed:
fix(abci): Return error when aggregate public key is not set

Fix ethereum tests

SatwikPrabhu committed:
Fix(BTE): Do not ignore error when making canonical block

SatwikPrabhu committed:
Chore(chain-spec): Remove todo comment testnet genesis hash

SatwikPrabhu committed:
fix: invalid pegin/pegout log should not be an error

arminsabouri committed:
feature(btc-server): make pegout id public

arminsabouri committed:
fix: use bitcoin::consenus::decode/encode for `bitcoin::BlockHash`

arminsabouri committed:
feature(btc-server): make psbt ext traits public

arminsabouri committed:
feature: bump rust-bitcoin 0.32.5 and corresponding libraries

arminsabouri committed:
chore(btc_server): clean up deprecated use of `txid()`

arminsabouri committed:
chore(btc_server): handle top level error conversion to `tonic::Status`

arminsabouri committed:
fix: frost ids should be generated from `u16`

arminsabouri committed:
test(btc_server): fix string matching in unit tests

arminsabouri committed:
chore(btc_server): remove unused template config file

arminsabouri committed:
refactor(btc_server): seperate bin and lib files

rwlockbg committed:
chore(repo): fix repo wide linting

And add coverage pins to the repo README.md

scottmillner committed:
fix(abci): log error when notifying pegins/pegouts fails instead of panic

arminsabouri committed:
ci: remove duplicate `--locked` param from codecov command

arminsabouri committed:
docs: update test suite docs with building instructions

scottmillner committed:
fix(abci): remove insert_block() and FCU

This work removes the BTE and beacon engine from PoA related code. Instead we write blocks straight into the the database.

rwlockbg committed:
fix(repo): Fixed failing db lock tests

rwlockbg committed:
fix(repo): Fixed panic with btc server metrics

SatwikPrabhu committed:
Refactor(btc-server): Make telemetry optional

Remove enable metrics flag

Refactor(btc-server): Make telemetry optional

Remove enable metrics flag

Address comments

Rebase changes

Refactor(btc-server): Make telemetry optional

Remove enable metrics flag

Address comments

Rebase changes

More rebase changes

Refactor(btc-server): Make telemetry optional

Remove enable metrics flag

Address comments

Rebase changes

More rebase changes

Remove typo

Fix crate error

Remove import mods

foramtting

foramtting

foramtting

Remove mut for variable

arminsabouri committed:
fix: frost ids should be generated from `u16`

arminsabouri committed:
chore: only print block number from canon state notif

arminsabouri committed:
fix(abci): commit results to database tx after insert

arminsabouri committed:
fix: prune the in-mem block cache after inserting block

arminsabouri committed:
fix(abci): only respond to commit after block insertion

arminsabouri committed:
refactor: use blockchain provider as canon state notifier

arminsabouri committed:
chore: bump to 1.0.3

arminsabouri committed:
test: pull app version from CARGO_PKG_VERSION

arminsabouri committed:
fix(abci): remove abci driver commit timeout

arminsabouri committed:
fix(abci): remove abci driver commit timeout

arminsabouri committed:
test(abci): remove un-needed commit test

[ticket created](https://github.com/botanix-labs/botanix/issues/907) to replace this coverage with something better

rwlockbg committed:
fix(repo): Updated crate versions

- updated clap version 4 -> 4.5
- updated once_cell 1.19 -> 1.20
- updated rustc-hash 2.0 -> 2.1
- updated thiserror 1.0 - > 2.0
- updated url 2.3 -> 2.5
- updated metrics 0.23 -> 0.24
- updated metrics-exporter-prometheus 0.15 -> 0.16
- updated metrics-process 2.1 -> 2.4
- updated metrics-util 0.17 -> 0.19
- updated tendermint-* 0.39 -> 0.40
- updated tokio tokio 1.39 -> 1.43
- updated tokio-stream 0.1.11 -> 0.1.17
- updated tokio-util 0.7.4 -> 0.7.13
- updated http 1.0 -> 1.2
- updated miniscript 11.0 -> 12.3
- updated bitcoin_hashes 0.14 -> 0.16
- updated thiserror 1.0 -> 2.0
- updated dirs 5.0 -> 6.0
- updated base64 0.21 -> 0.22
- updated arbitrary 1.3 -> 1.4
- updated iai-callgrind 0.11 -> 0.14
- updated pprof 0.13 -> 0.14
- updated proptest 1.4 -> 1.6
- updated serial_test 3 -> 3.2
- updated similar-asserts 1.5 -> 1.6
- updated tempfile 3.8 -> 3.16
- updated test-fuzz 5 -> 7

arminsabouri committed:
chore: bump to 1.0.4

arminsabouri committed:
feature(consensus): Support a generalized data parser with different compression strategies

arminsabouri committed:
refactor: pull trie updates from exec outcome

This commit also removes a whole bunch of unwraps from exectution utils.
And passes the db factory to execution utils instead of a provider

rwlockbg committed:
feat(abci): Added validate block check to the finalize endpoint

rwlockbg committed:
feat(ci): Added Fabio as codeowner for visibility

arminsabouri committed:
fix: re-order the numbers lock to match order in `remove_persisted_blocks`

This could lead to a deadlock senario where two threads of execution are
locked indefinetly waiting for the other to release their lock before
grabbing the next lock.

arminsabouri committed:
fix: re-order read locks in `in_memory` cache to match the order of which the write locks are obtained

arminsabouri committed:
chore: bump to 1.0.5

dependabot[bot] committed:
chore(deps): bump sigstore/cosign-installer from 3.7.0 to 3.8.0

Bumps [sigstore/cosign-installer](https://github.com/sigstore/cosign-installer) from 3.7.0 to 3.8.0.
- [Release notes](https://github.com/sigstore/cosign-installer/releases)
- [Commits](https://github.com/sigstore/cosign-installer/compare/v3.7.0...v3.8.0)

---
updated-dependencies:
- dependency-name: sigstore/cosign-installer
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

rwlockbg committed:
fix(ci): Adjusted coverage job to run once every day

arminsabouri committed:
test: clean up bitcoind working directory during clean up

arminsabouri committed:
chore: remove unused vars in auth builder

rwlockbg committed:
feat(repo): state sync (#506)

Co-authored-by: Scott Millner <22253830+scottmillner@users.noreply.github.com>
Co-authored-by: Armin Sabouri <armins88@gmail.com>

arminsabouri committed:
fix(poa): drop state sync provider db tx after using it

arminsabouri committed:
chore(poa): remove wallet state sync mod

arminsabouri committed:
test: use bitcoind rpc to generate and setup int. test wallet

arminsabouri committed:
test: clean up test suite resources after stopping signal handler thread

scottmillner committed:
feature(consensus): add pegins and pegouts to Commit message

scottmillner committed:
feature(consensus): remove btc-server from abci driver

scottmillner committed:
feature(consensus): unify pegins/pegouts/tx_index into one rpc endpoint

scottmillner committed:
feature(consensus): add test for new_consensus_checkpoint()

rwlockbg committed:
feat(node): Added state sync enabled boolean

scottmillner committed:
feature(consensus): refactor int tests to use unified checkpoint

arminsabouri committed:
feature(consensus): enforce coinbase maturity during pegin

arminsabouri committed:
test(peg_contract): clean up pk generation

arminsabouri committed:
test(peg_contract): assert correct amount

arminsabouri committed:
chore: remove unused health check task from authority/lib.rs

arminsabouri committed:
chore(authority): clean up imports that should be dev dependencies

arminsabouri committed:
chore(authority): remove deprecated wallet state sync

arminsabouri commented:
Replaced by #611
